pages/1.10/deploying-services/marathon-on-marathon/advanced/index.md

```diff
@@ -166,211 +186,156 @@ curl -i -k \
 
--  Delete the private key file from your file system to prevent bad actors from using the private key to authenticate to DC/OS.
 
-# Step 6 - Assign Permissions (Strict mode only)
-In this step, permissions are assigned to the Marathon-on-Marathon instance. Permissions are required in strict mode and are ignored in other security modes. 
+# Step 6: Install a Non-Native Marathon Instance with Assigned Role {.tabs}
+In this step, a non-native Marathon instance is installed on DC/OS with the Mesos role assigned.
 
-| Security Mode | Permissions |
-|---------------|----------------------|
-| Disabled | Not available |
-| Permissive | Not available |
-| Strict | Required |
+1.  Create a custom JSON config that will be used to install the custom non-native Marathon instance. The JSON file contents vary according to your [security mode](/1.10/security/ent/#security-modes).
 
-All CLI commands can also be executed via the [IAM API](/1.10/security/ent/iam-api).
+    Make sure you replace all the `${VARIABLES}` in the JSON file with the correct values for your test.
 
-1.  Grant the permission for user (`<service-account-id>`) with the `nobody` Linux user account specified. To use a different user account, replace `nobody` with the name of the user account. 
+    Alternatively, if you have exposed the values for all these variables in your terminal session (as explained in the [Variables Section](#variables-in-example)) you can use the following command to automatically replace all the variables with the correct values. It will save the result file in `marathon-filled.json`:
 
     ```bash
-    dcos security org users grant <service-account-id> dcos:mesos:master:task:user:nobody create --description "Allows Linux user nobody to execute tasks" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:framework:role:<myrole> create --description "Controls the ability of <myrole> to register as a framework with the Mesos master" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:reservation:role:<myrole> create --description "Controls the ability of <myrole> to reserve resources" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:volume:role:<myrole> create --description "Controls the ability of <myrole> to access volumes" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:reservation:principal:<service-account-id> delete --description "Controls the ability of <service-account-id> to reserve resources" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:task:app_id:/ create --description "Controls the ability to launch tasks" 
-    dcos security org users grant <service-account-id> dcos:mesos:master:volume:principal:<service-account-id> delete --description "Controls the ability of <service-account-id> to access volumes" 
+    perl -p -e 's/\$\{(\w+)\}/(exists $ENV{$1}?$ENV{$1}:"missing variable $1")/eg' < marathon.json > marathon-filled.json
```

Comment:
Why not just `envsubst`?

Reply:
Not available by default on mac... plus I don't want to replace all variables, only the ones following the `${VAR}` pattern

Reply:
What about the solution using Python suggested by @zen-dog?